[eas-cli] Automatically promote new cli release to staging on EAS Build (#1624)
expo/eas-cli: RELEASING.md

- If there are any entries in the "🛠 Breaking changes" section, bump the MAJOR version.
- Otherwise, if there are any entries in the "🎉 New features", bump the MINOR version.
- Otherwise, bump the PATCH version.

## Updating EAS CLI version on EAS Build

> Release process will automatically promote new version to staging. After verifying if it works there, you can promote it to production with `npm dist-tag add eas-cli@{version} latest-eas-build`.

Reviewer: ```suggestion
The release process promotes the new version to staging. After verifying it works there, you can promote it to production with `npm dist-tag add eas-cli@{version} latest-eas-build`.
```
